src/compiler/checker.ts

```diff
@@ -5478,8 +5478,8 @@ namespace ts {
             return type;
         }
 
-        function getJSDocTypeFromAssignmentDeclaration(declaredType: Type | undefined, expression: Expression, _symbol: Symbol, declaration: Declaration) {
-            const typeNode = getJSDocType(expression.parent);
+        function getAnnotatedTypeForAssignmentDeclaration(declaredType: Type | undefined, expression: Expression, symbol: Symbol, declaration: Declaration) {

```

Comment:
What `expression` gets passed to this function? The binary expression with the `=` operator? Or one side of it? Your change makes sense, but I’m trying to visualize what was happening before without actually running the code.

Reply:
Yep. Here is the relevant code that finds the expression from a JS declaration:

```ts
                const expression = (isBinaryExpression(declaration) || isCallExpression(declaration)) ? declaration :
                    isPropertyAccessExpression(declaration) ? isBinaryExpression(declaration.parent) ? declaration.parent : declaration :
                        undefined;
```

That chained-nested ternary is scary. Maybe we should figure out a way to unnest it.